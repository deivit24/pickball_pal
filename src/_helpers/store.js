import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
