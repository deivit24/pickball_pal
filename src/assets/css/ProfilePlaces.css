#ProfilePlaces {
  max-width: 600px;
  margin-bottom: 50px;
}

#ProfilePlaces h4 {
  font-weight: bold;
  color: #3f50b5;
  margin-bottom: 25px;
}

#ProfilePlaces .name {
  font-weight: bold;
  margin-bottom: 0px;
}
#ProfilePlaces .location {
  padding: 0px 10px 10px 10px;
}

#ProfilePlaces .name span {
  margin-right: 10px;
  color: rgb(255, 47, 47);
}
