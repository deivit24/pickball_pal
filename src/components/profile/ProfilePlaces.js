import React, { useState } from 'react';
import {
  GoogleMap,
  useLoadScript,
  Marker,
  InfoWindow,
} from '@react-google-maps/api';
import '../../assets/css/ProfilePlaces.css';
//GOOGLE MAPS SET UP
const GOOGLE_API = process.env.REACT_APP_GOOGLE_API;
const libraries = ['places'];

const mapContainerStyle = {
  width: '100%',
  height: '300px',
};
const ProfilePlaces = ({ places }) => {
  const [selected, setSelected] = useState(null);
  //GOOGLE MAPS
  const { isLoaded, loadError } = useLoadScript({
    googleMapsApiKey: GOOGLE_API,
    libraries,
  });

  if (loadError) return 'Error Loading Maps';
  if (!isLoaded) return 'Loading Maps';
  if (places.length > 0) {
    const center = {
      lat: places[0].lat,
      lng: places[0].lng,
    };
    return (
      <div id="ProfilePlaces" className="container">
        <div className="row">
          <div className="col-12">
            <h4>Places I play at</h4>
            <div className="row">
              {places.map((place) => (
                <div className="col-6 mb-3">
                  <p className="name">
                    <span>
                      <i className="fas fa-map-marker-alt"></i>
                    </span>
                    {place.name}
                  </p>
                  <p className="location">{place.location}</p>
                </div>
              ))}
            </div>

            <GoogleMap
              mapContainerStyle={mapContainerStyle}
              zoom={10}
              center={center}
            >
              {places.map((item) => (
                <div key={item.id}>
                  <Marker
                    position={{ lat: item.lat, lng: item.lng }}
                    icon={{ anchor: new window.google.maps.Point(15, 15) }}
                    onClick={() => {
                      setSelected(item);
                    }}
                  />
                </div>
              ))}
              {selected ? (
                <InfoWindow
                  position={{ lat: selected.lat, lng: selected.lng }}
                  onCloseClick={() => {
                    setSelected(null);
                  }}
                >
                  <div>
                    <p>{selected.location}</p>
                  </div>
                </InfoWindow>
              ) : null}
            </GoogleMap>
          </div>
        </div>
      </div>
    );
  }
  if (places.length === 0)
    return (
      <div id="ProfilePlaces" className="container">
        <div className="row">
          <div className="col-12">
            <h4>Places I play at</h4>
            <div className="row ">
              <div className="col-12 text-center">
                <h3>This player hasn't updated their locations</h3>
                <img
                  className="mx-auto"
                  src="https://cliply.co/wp-content/uploads/2019/03/371903340_LOCATION_MARKER_400.gif"
                  alt=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    );
};

export default ProfilePlaces;
