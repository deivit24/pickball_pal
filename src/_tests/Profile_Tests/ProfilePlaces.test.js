import React from 'react';
import { render } from '@testing-library/react';
import ProfilePlaces from '../../components/profile/ProfilePlaces';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing ProfilePlaces

describe('Testing the ProfilePlaces', () => {
  let places = {
    name: 'St Stephens',
    location: '100 St STephens rd Austin TX ',
    id: '1000',
    lng: 986794234,
    lat: 23424322,
  };

  it('it renders ProfilePlaces without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <ProfilePlaces places={places} />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <ProfilePlaces places={places} />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
