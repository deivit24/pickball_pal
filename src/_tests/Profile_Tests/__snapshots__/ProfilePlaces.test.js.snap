// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the ProfilePlaces matches snapshot 1`] = `
<DocumentFragment>
  Loading Maps
</DocumentFragment>
`;
